src/common/include/sirf/common/jcg.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <cmath>
+#include <iostream>
+
+#include <boost/shared_ptr.hpp>
+
+template<class vector>
+class anOperator {
```

Comment:
new base class shouldn't be defined in `jcg.h` but have its own file, and doxygen.

I'm not convinced about the name `anOperator`. I don't think other classes have `a` or `an` (although I'm not sure!). `Operator` seems better to me.

I guess in future have to think what to stick in here, but that can easily be postponed.

Reply:
done